* grobian@gentoo.org: on Solaris 10 _XOPEN_SOURCE_EXTENDED may not be
  defined when including wchar.h with g++

--- a/prog/check_funs.cpp
+++ b/prog/check_funs.cpp
@@ -18,10 +18,6 @@
 
 #include "settings.h"
 
-#ifdef DEFINE_XOPEN_SOURCE_EXTENDED
-#  define _XOPEN_SOURCE_EXTENDED 1
-#endif
-
 #ifdef CURSES_NON_POSIX
 #define CURSES_ONLY 1
 #endif
@@ -35,6 +31,10 @@
 
 #include "gettext.h"
 
+#ifdef DEFINE_XOPEN_SOURCE_EXTENDED
+#  define _XOPEN_SOURCE_EXTENDED 1
+#endif
+
 using namespace acommon;
 
 StackPtr<CheckerString> state;
